src/self-managed-debug/src/kubectl_port_forwarder.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Port forwards k8s service via Kubectl
+
+use anyhow::Result;
+use k8s_openapi::api::core::v1::Service;
+use kube::{Api, Client};
+
+use std::time::Duration;
+
+use mz_ore::retry::{self, RetryResult};
+use tracing::{error, info};
+
+use crate::SelfManagedDebugMode;
+#[derive(Debug, Clone)]
+pub struct KubectlPortForwarder {
+    pub namespace: String,
+    pub service_name: String,
+    pub target_port: i32,
+    pub local_address: String,
+    pub local_port: i32,
+    pub context: Option<String>,
+}
+
+impl KubectlPortForwarder {
```

Comment:
This is pretty much copied from the port_forwarding methods from system_catalog_dumper. Only thing that's changed is `SqlPortforwardingInfo` -> `KubetlPortForwarder`, converting the functions to methods, and everything else I mentioned in the commit message (i.e. getting rid of certain overrides).